src/transaction/log_recovery_analysis.cpp

```diff
@@ -2415,7 +2512,7 @@ log_recovery_analysis_from_trantable_snapshot (THREAD_ENTRY *thread_p,
     return true;
   });
 
-  LOG_SET_CURRENT_TRAN_INDEX (thread_p, sys_tran_index);
+  log_Gl.mvcc_table.complete_mvccids_if_still_active (LOG_SYSTEM_TRAN_INDEX, in_gaps_mvccids, true);
```

Comment:
I think here we need a TODO.
There could be transactions without logs encountered during recovery phase. If we complete there here we may encounter log records from these transactions later, during normal replication.